### ISSUE TITLE
LDAP Database: Can not connect

### ISSUE THREAD

**Issue by Bixilon** [anklimsk/wpkg-express-2#16]
In setup step 7 I only get a "Bad".

I can successfully connect to my ldap server with:
`ldapsearch -H ldap://192.168.200.6:389 -D "cn=wpkg,CN=Users,dc=domain,dc=de" -W -b "dc=domain,dc=de"`

What do I need to change in my Config file?
```
public $ldap = [
                'datasource' => 'CakeLdap.LdapExtSource',
                'persistent' => false,
                'host' => '192.168.200.6',
                'port' => 389,
                'login' => 'wpkg',
                'password' => '******',
                'database' => '',
                'basedn' => 'cn=wpkg,cn=Users,dc=domain,dc=de',
                'type' => 'ActiveDirectory',
                'tls' => false,
                'version' => 3,
        ];
```

I am using a samba4x server. Maybe you should give instructions in the Installation guide how to use the samba as ldap server...

Thanks

**Comment by anklimsk** [comment]
Hi! Try changing the 'type' => 'ActiveDirectory' parameter to 'Netscape' or 'OpenLDAP'.
I only tested on 'ActiveDirectory'.
Try turning on debugging [mode](https://github.com/anklimsk/wpkg-express-2/wiki/Enabling-debug-mode) and check the errors on the page and the file "app/tmp/logs/error.log".

**Comment by Bixilon** [comment]
Hey, thanks for the reply. Changed to Netscape, ... no success.

I can't even access the page, I need to configure the database first. If I want to omit ldap and configure no, the website tells me I must configure ldap.

I had debugging already enabled, no log messages. Only:
`2020-08-24 19:58:36 Ldap.error: Auth failed for 'wpkg'!`

**Comment by anklimsk** [comment]
Hi! Remove the "@" character in the file [app/Plugin/Datasources/Model/Datasource/LdapSource.php:261](https://github.com/anklimsk/wpkg-express-2/blob/v1.0.11/app/Plugin/Datasources/Model/Datasource/LdapSource.php#L261)
It looks like a bug in the connection configuration. Change your config to:
'login' => 'cn=wpkg,cn=Users,dc=domain,dc=de',
'basedn' => 'cn=Users,dc=domain,dc=de',

**Comment by Bixilon** [comment]
Yah, that did it. Thanks for the support

(now I am having many 404 errors for css and js files, like '/cake_theme/cache_css/css/CakeTheme.main.css'), but thats an other issue/misconfiguration.

I think I'll make an PR to improve the Installation Instructions. But before that, I will need it working :)

**Comment by anklimsk** [comment]
Good day. Fixed in a new release: [v1.0.12](https://github.com/anklimsk/wpkg-express-2/releases/tag/v1.0.12)
